tests/python/test_tools_smoke.py

```diff
@@ -0,0 +1,371 @@
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_tcpaccept(self):
+        self.run_with_int("tcpaccept.py")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_tcpconnect(self):
+        self.run_with_int("tcpconnect.py")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_tcpconnlat(self):
+        self.run_with_int("tcpconnlat.py")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_tcplife(self):
+        self.run_with_int("tcpconnlat.py")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_tcpretrans(self):
+        self.run_with_int("tcpretrans.py")
+
+    def test_tcptop(self):
+        self.run_with_duration("tcptop.py 1 1")
+
+    def test_tplist(self):
+        self.run_with_duration("tplist.py -p $(pgrep -n python)")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_trace(self):
+        self.run_with_int("trace.py SyS_open")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_ttysnoop(self):
+        self.run_with_int("ttysnoop.py /dev/console")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_ucalls(self):
+        # Unfortunately, it is not guaranteed that even the Python runtime
+        # has USDT probes, and indeed on the Ubuntu builtbot it doesn't. So
+        # we use the -S functionality for syscall tracing, which is largely
+        # superseded by the syscount tool.
+        self.run_with_int("ucalls.py -S $(pgrep -n python)")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_uflow(self):
+        # There is no suitable runtime to run this that we are guaranteed to
+        # have, so don't test.
+        # self.run_with_int("uflow.py python $(pgrep -n python)")
+        pass
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_ugc(self):
+        # This requires a runtime that has GC probes to be installed.
+        # Python has them, but only in very recent versions. Skip.
+        pass
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_uobjnew(self):
+        self.run_with_int("uobjnew.py c $(pgrep -n python)")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_ustat(self):
+        self.run_with_duration("ustat.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_uthreads(self):
+        self.run_with_int("uthreads.py $(pgrep -n python)")
+
+    def test_vfscount(self):
+        self.run_with_int("vfscount.py")
+
+    def test_vfsstat(self):
+        self.run_with_duration("vfsstat.py 1 1")
+
+    def test_wakeuptime(self):
+        self.run_with_duration("wakeuptime.py 1")
+
+    def test_xfsdist(self):
+        # Doesn't work on build bot because xfs functions not present in the
```

Comment:
I'd really need access to the bots to debug this then I'm afraid. I can't go on on much more than the log outputs ...